catalog/async/catalog-async-data-api/src/main/java/org/codice/ddf/catalog/async/data/api/internal/ProcessResource.java

```diff
@@ -93,4 +99,16 @@
    * <p>Look into having the ProcessResource implement InputStream
    */
   void close();
+
+  /**
+   * Convenience method to attempt to read the contents of the <code>InputStream</code> into a
+   * <code>byte</code> array.
+   *
+   * <p>Note that {@link #getByteArray()} should be idempotent if {@link #getInputStream()} has not
+   * been initially invoked.
+   *
+   * @return byte[] representation of the input stream
+   * @throws IOException if the stream can not be read into the array
+   */
+  byte[] getByteArray() throws IOException;
```

Comment:
❗️ `byte[]` might not be a good candidate as a return type. Either the implementation would have to clone the array which means more memory or again it would (has done below) expose the internal buffer allowing it to be changed without its knowledge.

How about a ByteBuffer in read-only mode instead?